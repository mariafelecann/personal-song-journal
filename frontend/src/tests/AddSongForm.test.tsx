// // describe('AddSong', () => {
// //     it('renders the form and submits data', () => {
// //         const onBackButtonClick = vi.fn();
// //         const onSubmitHandler = vi.fn();

// import {render} from '@testing-library/react';
// import {vi} from 'vitest';
// import AddSong from '../components/AddSongForm';

// //         render(
// //             <AddSong
// //                 onBackButtonClick={onBackButtonClick}
// //                 onSubmitHandler={onSubmitHandler}
// //             />,
// //         );

// //         // Check if the form renders
// //         expect(screen.getByText('Add a new Song!'));

// //         // Fill in the form fields
// //         fireEvent.change(screen.getByLabelText('Title:'), {
// //             target: {value: 'Test Song'},
// //         });
// //         fireEvent.change(screen.getByLabelText('Artist:'), {
// //             target: {value: 'Test Artist'},
// //         });
// //         fireEvent.change(screen.getByLabelText('Rating:'), {
// //             target: {value: '5'},
// //         });
// //         fireEvent.change(screen.getByLabelText('Review:'), {
// //             target: {value: 'Great song!'},
// //         });

// //         // Submit the form
// //         fireEvent.click(screen.getByText('Add Song'));

// //         // Check if onSubmitHandler was called with the correct data
// //         expect(onSubmitHandler).toHaveBeenCalledWith({
// //             id: expect.any(String),
// //             title: 'Test Song',
// //             artist: 'Test Artist',
// //             rating: 5,
// //             review: 'Great song!',
// //         });
// //     });
// // });

// // describe('AddSong', () => {
// //     it('renders the form and submits data', async () => {
// //         const onBackButtonClick = vi.fn();
// //         const onSubmitHandler = vi.fn();

// //         render(
// //             <AddSong
// //                 onBackButtonClick={onBackButtonClick}
// //                 onSubmitHandler={onSubmitHandler}
// //             />,
// //         );

// //         // Check if the form renders
// //         expect(screen.getByText('Add a new Song!')).not.toBeNull();

// //         // Fill in the form fields
// //         userEvent.type(screen.getByLabelText('Title:'), 'Test Song');
// //         userEvent.type(screen.getByLabelText('Artist:'), 'Test Artist');
// //         userEvent.type(screen.getByLabelText('Rating:'), '5');
// //         userEvent.type(screen.getByLabelText('Review:'), 'Great song!');

// //         // Submit the form
// //         userEvent.click(screen.getByText('Add Song'));

// //         // Wait for any asynchronous updates to complete
// //         await vi.nextTick();

// //         // Check if onSubmitHandler was called with the correct data
// //         expect(onSubmitHandler).toHaveBeenCalledWith({
// //             id: expect.any(String),
// //             title: 'Test Song',
// //             artist: 'Test Artist',
// //             rating: 5,
// //             review: 'Great song!',
// //         });
// //     });
// // });

// // describe('AddSong', () => {
// //     it('renders the form and submits data', async () => {
// //         const onBackButtonClick = vi.fn();
// //         const onSubmitHandler = vi.fn();

// //         render(
// //             <AddSong
// //                 onBackButtonClick={onBackButtonClick}
// //                 onSubmitHandler={onSubmitHandler}
// //             />,
// //         );
// //         expect(screen.getAllByText('Add a new Song!')).not.toBeNull();

// //         userEvent.type(screen.getByLabelText('Title:'), 'Test Song');
// //         userEvent.type(screen.getByLabelText('Artist:'), 'Test Artist');
// //         userEvent.type(screen.getByLabelText('Rating:'), '5');
// //         userEvent.type(screen.getByLabelText('Review:'), 'Great song!');

// //         userEvent.click(screen.getByText('Add Song'));

// //         await waitFor(() => {
// //             expect(onSubmitHandler).toHaveBeenCalledWith({
// //                 id: expect.any(String),
// //                 title: 'Test Song',
// //                 artist: 'Test Artist',
// //                 rating: 5,
// //                 review: 'Great song!',
// //             });
// //         });
// //     });
// // });
// import {describe, expect, it} from 'vitest';

// describe('AddSong', () => {
//     it('renders the form and submits data', async () => {
//         const onBackButtonClick = vi.fn();
//         const onSubmitHandler = vi.fn();

//         const renderedComp = render(
//             <AddSong
//                 onBackButtonClick={onBackButtonClick}
//                 onSubmitHandler={onSubmitHandler}
//             />,
//         );
//         expect(renderedComp).toMatchSnapshot();
//     });
// });
