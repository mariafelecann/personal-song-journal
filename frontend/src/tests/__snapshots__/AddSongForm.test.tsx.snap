// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AddSong > renders the form and submits data 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        class="add-song-form-container"
      >
        <header>
          <h3>
            Add a new Song!
          </h3>
        </header>
        <div
          class="elements-container"
        >
          <div>
            <label>
              Title:
            </label>
            <input
              class="input-box"
              type="text"
              value=""
            />
          </div>
          <div>
            <label>
              Artist:
            </label>
            <input
              class="input-box"
              type="text"
              value=""
            />
          </div>
          <div>
            <label>
              Rating:
            </label>
            <input
              class="input-box"
              type="text"
              value=""
            />
          </div>
          <div>
            <label>
              Review:
            </label>
            <input
              class="input-box"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="buttons-container"
        >
          <button>
            Back
          </button>
          <button
            type="submit"
          >
            Add Song
          </button>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      class="add-song-form-container"
    >
      <header>
        <h3>
          Add a new Song!
        </h3>
      </header>
      <div
        class="elements-container"
      >
        <div>
          <label>
            Title:
          </label>
          <input
            class="input-box"
            type="text"
            value=""
          />
        </div>
        <div>
          <label>
            Artist:
          </label>
          <input
            class="input-box"
            type="text"
            value=""
          />
        </div>
        <div>
          <label>
            Rating:
          </label>
          <input
            class="input-box"
            type="text"
            value=""
          />
        </div>
        <div>
          <label>
            Review:
          </label>
          <input
            class="input-box"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="buttons-container"
      >
        <button>
          Back
        </button>
        <button
          type="submit"
        >
          Add Song
        </button>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
